lib/grape/dsl/inside_route.rb

```diff
@@ -6,46 +6,73 @@ module InsideRoute
+      #   position in the filter evaluation sequence denoted by +type+.  This
+      #   defaults to an empty module if no overrides are defined for the given
+      #   filter +type+.
+      def self.post_filter_methods(type)
+        @post_filter_modules ||= { before: PostBeforeFilter }
+        @post_filter_modules[type] || Module.new
+      end
 
-        fail ArgumentError, 'Tried to filter for declared parameters but none exist.' unless declared_params
+      # Methods which should not be available in filters until the before filter
+      # has completed
+      module PostBeforeFilter
+        def declared(params, options = {}, declared_params = nil)
+          options = options.reverse_merge(include_missing: true, include_parent_namespaces: true)
 
-        if params.is_a? Array
-          params.map do |param|
-            declared(param || {}, options, declared_params)
-          end
-        else
-          declared_params.each_with_object(Hashie::Mash.new) do |key, hash|
-            key = { key => nil } unless key.is_a? Hash
+          declared_params ||= (!options[:include_parent_namespaces] ? route_setting(:declared_params) : (route_setting(:saved_declared_params) || [])).flatten(1) || []
 
-            key.each_pair do |parent, children|
-              output_key = options[:stringify] ? parent.to_s : parent.to_sym
+          fail ArgumentError, 'Tried to filter for declared parameters but none exist.' unless declared_params
+
+          if params.is_a? Array
+            params.map do |param|
+              declared(param || {}, options, declared_params)
+            end
+          else
+            declared_params.each_with_object(Hashie::Mash.new) do |key, hash|
+              key = { key => nil } unless key.is_a? Hash
 
-              next unless options[:include_missing] || params.key?(parent)
+              key.each_pair do |parent, children|
+                output_key = options[:stringify] ? parent.to_s : parent.to_sym
 
-              hash[output_key] = if children
-                                   children_params = params[parent] || (children.is_a?(Array) ? [] : {})
-                                   declared(children_params, options, Array(children))
-                                 else
-                                   params[parent]
-                                 end
+                next unless options[:include_missing] || params.key?(parent)
+
+                hash[output_key] = if children
+                                     children_params = params[parent] || (children.is_a?(Array) ? [] : {})
+                                     declared(children_params, options, Array(children))
+                                   else
+                                     params[parent]
+                                   end
+              end
             end
           end
         end
       end
 
+      # A filtering method that will return a hash
```

Comment:
It's a little weird to have the docs on the method stub rather than the full version in the `PostBeforeFilter` module.  Anyone more savvy with YARD know if this is the best way to be doing this?